grammar/ja/src/rules.rs

```diff
@@ -983,20 +980,25 @@ pub fn rules_time(b: &mut RuleSetBuilder<Dimension>) -> RustlingResult<()> {
         time_check!(form!(Form::Year(_))),
         time_check!(form!(Form::MonthDay(_))),
         |year, month_day| {
-            year.value().intersect(&month_day.value())
+            Ok(year.value().intersect(&month_day.value())?
+              .form(Form::YearMonthDay(None)))
         }
     );
     b.rule_2("hour",
-                      integer_check_by_range!(0, 24),
-                      b.reg(r#"時"#)?,
-                      |a, _| helpers::hour(a.value().value as u32, true)
-    );
-    b.rule_4("<hour and minutes",
-                      integer_check_by_range!(0, 24),
-                      b.reg(r#"時"#)?,
-                      integer_check_by_range!(0, 59),
-                      b.reg(r#"分"#)?,
-                      |a, _, b, _| helpers::hour_minute(a.value().value as u32, b.value().value as u32, true)
+        integer_check_by_range!(0, 24),
+        b.reg(r#"時"#)?,
+        |a, _| helpers::hour(a.value().value as u32, true)
+    );
+    b.rule_4("hour and minutes",
+        integer_check_by_range!(0, 24),
```

Comment:
It should be integer_check_by_range!(0, 23) as there can't be minutes after 24h in Japanese

Reply:
done!